### PR TITLE
Big-endian test case fixes: System.Runtime

### PR DESCRIPTION
**Pull request by uweigand**
* Byte-swap input/output for BitConverter tests

* Update byte index for GetByte/SetByte tests on big-endian systems

### CODE REVIEW

File: src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs [hunk 1]
```diff
@@ -46,131 +46,231 @@ public override void ConvertFromBool(bool boolean, byte[] expected)
 
         public override void ConvertFromShort(short num, byte[] expected)
         {
+            if (!BitConverter.IsLittleEndian)
+            {
+                expected = (byte[]) expected.Clone();
+                Array.Reverse(expected);
+            }
```
Comment (stephentoub) [inline src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs:53]: Did you consider extracting this into a helper?

File: src/libraries/System.Runtime/tests/System/BufferTests.cs [hunk 2]
```diff
@@ -91,6 +91,10 @@ public static void ByteLength_Invalid()
         [InlineData(new uint[] { 0x01234567, 0x89abcdef }, 7, 0x89)]
         public static void GetByte(Array array, int index, int expected)
         {
+            if (!BitConverter.IsLittleEndian)
+            {
+                index = index ^ 3;
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/BufferTests.cs:96]: What is this doing?

File: src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs (hunk 1 above)
Comment (uweigand) [reply src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs:53]: OK, I'll do that.

File: src/libraries/System.Runtime/tests/System/BufferTests.cs (hunk 2 above)
Comment (uweigand) [reply src/libraries/System.Runtime/tests/System/BufferTests.cs:96]: The input is an array of uint, which is accessed via a byte index.  To compensate for byte order, this patch simply updates the index to point to the "swapped" position (0 -> 3, 1 -> 2, 2 -> 1, 3 -> 0, 4 -> 7, 5 -> 6, 6 -> 5, 7 -> 4 etc.).  This index transformation can be implemented via the ^ 3 operation.

File: src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs (hunk 1 above)
Comment (uweigand) [reply src/libraries/System.Runtime.Extensions/tests/System/BitConverterSpan.cs]: Patch updated.

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**